client/client_test.go

```diff
@@ -1238,6 +1248,48 @@ func testListTarget(t *testing.T, rootType string) {
 	require.True(t, reflect.DeepEqual(*currentTarget, newCurrentTarget.Target), "current target does not match")
 }
 
+func testListTargetWithCustom(t *testing.T, rootType string) {
```

Comment:
Non-blocking:  I wonder if it'd make sense to just fold this into `testListTarget` by adding custom metadata to one of the targets (either `latest` or `custom`)?